tools/Import/Importer.cs

```diff
@@ -0,0 +1,271 @@
+        private static readonly int BlobRangeSize = int.Parse(ConfigurationManager.AppSettings["BlobRangeSize"]);
+        private static readonly int ReadThreads = int.Parse(ConfigurationManager.AppSettings["ReadThreads"]);
+        private static readonly int WriteThreads = int.Parse(ConfigurationManager.AppSettings["WriteThreads"]);
+        private static readonly int NumberOfBlobsToSkip = int.Parse(ConfigurationManager.AppSettings["NumberOfBlobsToSkip"]);
+        private static readonly int NumberOfBlobsToImport = int.Parse(ConfigurationManager.AppSettings["NumberOfBlobsToImport"]);
+        private static readonly int ReportingPeriodSec = int.Parse(ConfigurationManager.AppSettings["ReportingPeriodSec"]);
+        private static readonly int MaxRetries = int.Parse(ConfigurationManager.AppSettings["MaxRetries"]);
+        private static readonly bool UseStringJsonParser = bool.Parse(ConfigurationManager.AppSettings["UseStringJsonParser"]);
+        private static readonly bool UseStringJsonParserCompare = bool.Parse(ConfigurationManager.AppSettings["UseStringJsonParserCompare"]);
+
+        private static long readers = 0L;
+        private static long resourcesRead = 0L;
+        private static Stopwatch swReads = Stopwatch.StartNew();
+
+        internal static void Run()
+        {
+            ServicePointManager.DefaultConnectionLimit = 2048;
+
+            if (string.IsNullOrEmpty(fhirEndpoint2))
+            {
+                fhirEndpoint2 = FhirEndpoint;
+            }
+
+            var globalLogPrefix = $"GlobalBlobRange=[{NumberOfBlobsToSkip + 1}-{NumberOfBlobsToImport}]";
+            Console.WriteLine($"{globalLogPrefix}: Starting...");
+            var blobContainerClient = GetContainer(ConnectionString, ContainerName);
+            var blobs = blobContainerClient.GetBlobs().Skip(NumberOfBlobsToSkip).Take(NumberOfBlobsToImport - NumberOfBlobsToSkip).ToList();
+            var writers = 0L;
+            var epCalls = 0L;
+            var waits = 0L;
+            var globalResourceCount = 0L;
+            var swWrites = Stopwatch.StartNew();
+            var swReport = Stopwatch.StartNew();
+            var swReportReads = Stopwatch.StartNew();
+            BatchExtensions.ExecuteInParallelBatches(blobs, ReadThreads, BlobRangeSize, (reader, blobRangeInt) =>
+            {
+                var localSw = Stopwatch.StartNew();
+                var localResourceCount = 0L;
+                var blobRangeIndex = blobRangeInt.Item1;
+                var blobRange = blobRangeInt.Item2;
+                var localLogPrefix = $"Reader={reader}.BlobRange=[{NumberOfBlobsToSkip + (blobRangeIndex * BlobRangeSize) + 1}-{NumberOfBlobsToSkip + (blobRangeIndex * BlobRangeSize) + blobRange.Count}]";
+                Console.WriteLine($"{localLogPrefix}: Starting...");
+
+                BatchExtensions.ExecuteInParallelBatches(GetLinesInBlobRange(blobRange, localLogPrefix), WriteThreads / ReadThreads, 100, (thread, lineBatch) =>
```

Comment:
Is the number of write threads per read thread (or within a read thread)? Is 100 the write batch size? Any significance of WriteThreads / ReadThreads?

Reply:
Write threads is total. Each "reader" gets its equal share of writers. 100 is batch size in internal queueing, all http calls are still single resource calls.